react/features/notifications/components/AbstractNotificationsContainer.js

```diff
@@ -0,0 +1,155 @@
+         * notification. This value is set on the instance and not state to
+         * avoid additional re-renders.
+         *
+         * @type {number|null}
+         */
+        this._notificationDismissTimeout = null;
+
+        // Bind event handlers so they are only bound once for every instance.
+        this._onDismissed = this._onDismissed.bind(this);
+    }
+
+    /**
+     * Sets a timeout if the currently displayed notification has changed.
+     *
+     * @inheritdoc
+     */
+    componentDidUpdate(prevProps: P) {
+        const { _notifications, _showNotifications } = this.props;
+
+        if (_notifications.length) {
+            const notification = _notifications[0];
+            const previousNotification
+                = prevProps._notifications.length
+                    ? prevProps._notifications[0]
+                    : undefined;
+
+            if (notification !== previousNotification) {
+                this._notificationDismissTimeout
+                    && clearTimeout(this._notificationDismissTimeout);
+
+                this._notificationDismissTimeout = null;
+
+                if (_showNotifications && notification) {
+                    const { timeout, uid } = notification;
+
+                    this._notificationDismissTimeout = setTimeout(() => {
+                        // Perform a no-op if a timeout is not specified.
+                        if (Number.isInteger(timeout)) {
+                            this._onDismissed(uid);
+                        }
+                    }, timeout);
+                }
+            }
+        }
+    }
+
+    /**
+     * Clear any dismissal timeout that is still active.
+     *
+     * @inheritdoc
+     * returns {void}
+     */
+    componentWillUnmount() {
+        this._notificationDismissTimeout
+            && clearTimeout(this._notificationDismissTimeout);
+    }
+
+    _onDismissed: number => void;
+
+    /**
+     * Emits an action to remove the notification from the redux store so it
+     * stops displaying.
+     *
+     * @param {number} uid - The id of the notification to be removed.
+     * @private
+     * @returns {void}
+     */
+    _onDismissed(uid) {
+        this._notificationDismissTimeout
+            && clearTimeout(this._notificationDismissTimeout);
+        this._notificationDismissTimeout = null;
+
+        this.props.dispatch(hideNotification(uid));
+    }
+}
+
+/**
+ * Maps (parts of) the Redux state to the associated NotificationsContainer's
+ * props.
+ *
+ * @param {Object} state - The Redux state.
+ * @private
+ * @returns {{
+ *     _notifications: Array,
+ *     _showNotifications: boolean
+ * }}
+ */
+export function _abstractMapStateToProps(state: Object) {
+    // TODO: Per existing behavior, notifications should not display when an
+    // overlay is visible. This logic for checking overlay display can likely be
+    // simplified.
+    const {
+        connectionEstablished,
+        haveToReload,
```

Comment:
hmm I don't see neither 'haveToReload' nor 'connectionEstablished' defined on the current master... what am I missing ?

Reply:
I don't see 'haveToReload' nor 'connectionEstablished' fields here:
https://github.com/jitsi/jitsi-meet/blob/master/react/features/overlay/reducer.js
Was this added as part of this PR or is there magic somewhere ?